### ISSUE TITLE
ownCloud Container will not start after containers were upgraded

### ISSUE THREAD

**Issue by BzowK** [owncloud-docker/server#127]
Hey Guys - 

I use WatchTower to keep my containers up to date and this evening it upgraded my 3 ownCloud ones including redis:latest, mariadb:latest, & owncloud/server:10.0.  Once updated, they came back up, but after a few seconds the owncloud/server:10.0 one went back offline.

The logs in it show the following:

> eating volume folders...
> Creating hook folders...
> Waiting for MySQL...
> wait-for-it: waiting 180 seconds for db:3306
> wait-for-it: db:3306 is available after 0 seconds
> Waiting for Redis...
> wait-for-it: waiting 60 seconds for redis:6379
> wait-for-it: redis:6379 is available after 0 seconds
> Writing config file...
> Fixing base perms...
> Fixing data perms...
> Fixing hook perms...
> Upgrading server database...
> ownCloud or one of the apps require upgrade - only a limited number of commands are available
> You may use your browser or the occ upgrade command to do the upgrade
> 2019-05-04T02:58:20+00:00 Set log level to debug
> 2019-05-04T02:58:20+00:00 Repair step: Repair MySQL database engine
> 2019-05-04T02:58:20+00:00 Repair step: Repair MySQL collation
> 2019-05-04T02:58:20+00:00 Repair info: All tables already have the correct collation -> nothing to do
> 2019-05-04T02:58:20+00:00 Repair step: Repair SQLite autoincrement
> 2019-05-04T02:58:20+00:00 Repair step: Repair orphaned reshare
> 2019-05-04T02:58:20+00:00 Repair step: Repair duplicate entries in oc_lucene_status
> 2019-05-04T02:58:20+00:00 Repair info: lucene_status table does not exist -> nothing to do
> 2019-05-04T02:58:20+00:00 Repair step: Upgrade app code from the marketplace
> 2019-05-04T02:58:20+00:00 Repair warning: You have incompatible or missing apps enabled that could not be found or updated via the marketplace.
> 2019-05-04T02:58:20+00:00 Repair warning: Please install or update the following apps manually or disable them with:
> occ app:disable sensorlogger
> 2019-05-04T02:58:20+00:00 Repair warning: For manually updating, see https://doc.owncloud.org/server/10.0/go.php?to=admin-marketplace-apps
> 2019-05-04T02:58:20+00:00 OC\RepairException: Upgrade is not possible
> 2019-05-04T02:58:20+00:00 Update failed
> 2019-05-04T02:58:20+00:00 Maintenance mode is kept active
> 2019-05-04T02:58:20+00:00 Reset log level

The only thing that stands out to me is to disable sensorlogger which I have no problem doing, but not sure how seeing as since the container only stays up for a few seconds, I can't get to a bash prompt.  I looked through the mounted volumes to see if the apps were in there, but only found a few sensorlogger files for sql.  I tried moving them temporarilly, but still got the same error.

Any suggestions for how to get back up again and/or disable that app assuming it's the only thing standing in my way?  I considered changing maintenance mode to "false" in config.php, but not even getting maintenance mode ui - the container just stops after starting.

Thanks!
 
Note: I'm not 100% the exact version prior to container upgrade, but know it was a recent 10.x one.  

**config.php**

```
<?php
$CONFIG = array (
  'apps_paths' => 
  array (
    0 => 
    array (
      'path' => '/var/www/owncloud/apps',
      'url' => '/apps',
      'writable' => false,
    ),
    1 => 
    array (
      'path' => '/var/www/owncloud/custom',
      'url' => '/custom',
      'writable' => true,
    ),
  ),
  'trusted_domains' => 
  array (
    0 => 'localhost',
  ),
  'datadirectory' => '/mnt/data/files',
  'dbtype' => 'mysql',
  'dbhost' => 'db:3306',
  'dbname' => 'owncloud',
  'dbuser' => 'owncloud',
  'dbpassword' => 'xxxxxxxxx',
  'dbtableprefix' => 'oc_',
  'log_type' => 'owncloud',
  'supportedDatabases' => 
  array (
    0 => 'sqlite',
    1 => 'mysql',
    2 => 'pgsql',
  ),
  'upgrade.disable-web' => true,
  'default_language' => 'en',
  'overwrite.cli.url' => 'http://localhost/owncloud',
  'htaccess.RewriteBase' => '/owncloud',
  'logfile' => '/mnt/data/files/owncloud.log',
  'loglevel' => 2,
  'memcache.local' => '\\OC\\Memcache\\APCu',
  'filelocking.enabled' => true,
  'memcache.distributed' => '\\OC\\Memcache\\Redis',
  'memcache.locking' => '\\OC\\Memcache\\Redis',
  'redis' => 
  array (
    'host' => 'redis',
    'port' => '6379',
  ),
  'passwordsalt' => 'xxxxxxxxxxxxxxxxxxxx',
  'secret' => 'xxxxxxxxxxxxxxxxxxxxxxxx',
  'version' => '10.0.10.4',
  'logtimezone' => 'UTC',
  'installed' => true,
  'instanceid' => 'xxxxxxxxxx',
  'maintenance' => true,
);

```

**Comment by tboerger** [comment]
This is a bad corner case that could happen. You got 3 options in that case:

1) disable the app within the database 
2) execute docker exec owncloud_container_name occ app:disable sensorlogger while starting the container really often until it succeeds 
3) inject a shell script into /etc/pre_install.d as volume which disables the app

**Comment by BzowK** [comment]
Appreciate your reply and apologies for my late one - been busy :)

I think I'm back up!  Below is what I ended up doing...

I  decided to attempt your 2nd suggestion first as don't have too much CLI experience with mySQL.  I connected to the host via SSH, queued the suggested command string, started the container via Portainer, then executed the string soon afterwards.  The terminal didn't echo anything for about 20 seconds, then finally stated:
> ownCloud or one of the apps require upgrade - only a limited number of commands are available
> You may use your browser or the occ upgrade command to do the upgrade
> sensorlogger disabled

After a couple of minutes, I restarted the container and found it was in maintenance mode which I reverted via config.php and was finally back in.  Unfortunately, I had two files that failed hash - .htaccess & .user.ini.  As Settings showed I was on 10.0.10.4, I downloaded it's tarball, extracted into Windows, then copied those two files into a folder which was mapped to a volume on the container.  

I then connected to the container's bash, renamed the two old files (above) in /var/www/owncloud, copied the uploaded ones from the mapped volume folder into /var/www/owncloud, and rescanned the files from Settings.  Once done, no more errors!

Appreciate the help!

Thanks

**Comment by tboerger** [comment]
The sum issue is an upstream issue, so I gonna close this issue now.

**Comment by McFrappe** [comment]
> This is a bad corner case that could happen. You got 3 options in that case:
> 
> 1. disable the app within the database
> 2. execute docker exec owncloud_container_name occ app:disable sensorlogger while starting the container really often until it succeeds
> 3. inject a shell script into /etc/pre_install.d as volume which disables the app

Just wanted to thank you for this. I think this shows that I need to learn more about docker. 

context: I rebooted my server and the owncloud_server instance was just not booting. I checked the logs and saw that the app ''gallery'' was the culprit and needed to disable it. Okay, lets try to enter the container and disable it. Unable to. The container was continuously restarting and I could not execute `docker exec -it CONTAINER_NAME bash`, as the container never starts. But, I was able to disable the app gallery by running:

`Docker stop owncloud_server && docker exec owncloud_server occ app:disable gallery`.
